### PR TITLE
feat(utils): Add metrics to circuit breaker

### PR DESCRIPTION
**Pull request by lobsterkatie**
This adds two metrics to the circuit breaker, one which fires when the breaker is tripped (with a 100% sample rate, since this shouldn't happen often and we always want to know when it does) and one which fires when a request is blocked. Each metric is namespaced by the circuit breaker's key, so different circuit breaker instances can be monitored separately.

### CODE REVIEW

File: src/sentry/utils/circuit_breaker2.py [hunk 1]
```diff
@@ -240,6 +244,11 @@ def record_error(self) -> None:
                     "error_limit_window": controlling_quota.window_seconds,
                 },
             )
+            metrics.incr(
+                f"circuit_breaker.{self.key}.error_limit_hit",
+                sample_rate=1.0,
+                tags={"current_state": state},
```
Comment (jangjodi) [inline src/sentry/utils/circuit_breaker2.py:250]: same comment as on the previous PR - I believe we should be able to use `state.value` to get the string value here

Comment (lobsterkatie) [reply src/sentry/utils/circuit_breaker2.py]: Yup, good call. Done.

File: tests/sentry/utils/test_circuit_breaker2.py [hunk 2]
```diff
@@ -432,6 +435,11 @@ def test_handles_hitting_max_errors_in_non_broken_state(self, mock_logger: Magic
                     "error_limit_window": config["error_limit_window"],
                 },
             )
+            mock_metrics_incr.assert_called_with(
+                "circuit_breaker.dogs_are_great.error_limit_hit",
+                sample_rate=1.0,
+                tags={"current_state": state},
```
Comment (jangjodi) [inline tests/sentry/utils/test_circuit_breaker2.py:441]: should this also be `state.value`?

**Review by jangjodi — APPROVED** [review]

File: tests/sentry/utils/test_circuit_breaker2.py (hunk 2 above)
Comment (lobsterkatie) [reply tests/sentry/utils/test_circuit_breaker2.py:441]: Dammit, LOL! Yes it should. Good catch!

Comment (lobsterkatie) [reply tests/sentry/utils/test_circuit_breaker2.py:441]: Wait, no, that's the one I fixed. 🤔

**Pull request merged**